sympy/sets/sets.py

```diff
@@ -584,7 +584,7 @@ def _eval_Eq(self, other):
         if len(self.args) != len(other.args):
             return false
 
-        return And(*map(lambda x, y: Eq(x, y), self.args, other.args))
+        return And(*[Eq(x, y) for x, y in zip(self.args, other.args)])
```

Comment:
Good point. There are other list compressions here which use `*` on lists. There are nearly 13 such instances. Can I change them in this PR itself ?